test/controllers/ios_app/sensors_controller_test.rb

```diff
@@ -0,0 +1,74 @@
+require 'test_helper'
+
+def parsed_response
+  JSON.parse(response.body)
+end
+
+module IosApp
+  class SensorsControllerTest < ActionDispatch::IntegrationTest
+    setup do
+      @sensor = create(:sensor)
+    end
+
+    test 'should get index with latest measurement' do
+      @sensor.measurements << create(:measurement, temperature: 20)
+
+      get ios_app_sensors_url, env: public_auth_header
```

Comment:
Just out of interest, where is this variable(?) defined?

Reply:
What is "this"?

You could write this line as `get(ios_app_sensors_url(), { env: public_auth_header() })`

* `get` is a Rails testing helper method 
* `ios_app_sensors_url` is a method generated during runtime by the routing mechanism
*  `public_auth_header` is a custom helper method which I defined

Reply:
Ah, this part is what I was looking for:

> ios_app_sensors_url is a method generated during runtime by the routing mechanism

Rails magic at work again :smile: Thanks for the explanation!